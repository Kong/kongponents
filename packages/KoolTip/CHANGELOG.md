# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.1.5 (2021-11-17)

**Note:** Version bump only for package @kongponents/kooltip





## 6.1.4 (2021-11-17)


### Bug Fixes

* general fixes ([#415](https://github.com/Kong/kongponents/issues/415)) ([5b4bb48](https://github.com/Kong/kongponents/commit/5b4bb484d74e75779d39cf901b2cdaa23818e8e7))





## 6.1.3 (2021-11-17)

**Note:** Version bump only for package @kongponents/kooltip





## [6.1.2](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@5.0.8...@kongponents/kooltip@6.1.2) (2021-11-17)

Kongponents redesign - no changes to this component


## [5.0.8](https://github.com/Kong/kongponents/compare/v5.0.7...v5.0.8) (2021-11-15)

**Note:** Version bump only for package @kongponents/kooltip





## [5.0.7](https://github.com/Kong/kongponents/compare/v5.0.6...v5.0.7) (2021-10-26)

**Note:** Version bump only for package @kongponents/kooltip





## [5.0.6](https://github.com/Kong/kongponents/compare/v5.0.5...v5.0.6) (2021-10-26)

**Note:** Version bump only for package @kongponents/kooltip





## [5.0.5](https://github.com/Kong/kongponents/compare/v5.0.4...v5.0.5) (2021-10-13)

**Note:** Version bump only for package @kongponents/kooltip





## [5.0.4](https://github.com/Kong/kongponents/compare/v5.0.3...v5.0.4) (2021-10-06)

**Note:** Version bump only for package @kongponents/kooltip





## [5.0.3](https://github.com/Kong/kongponents/compare/v5.0.2...v5.0.3) (2021-09-22)

**Note:** Version bump only for package @kongponents/kooltip





## [5.0.2](https://github.com/Kong/kongponents/compare/v5.0.1...v5.0.2) (2021-09-15)

**Note:** Version bump only for package @kongponents/kooltip





## [5.0.1](https://github.com/Kong/kongponents/compare/v5.0.0...v5.0.1) (2021-09-13)

**Note:** Version bump only for package @kongponents/kooltip





# [5.0.0](https://github.com/Kong/kongponents/compare/v4.0.2...v5.0.0) (2021-09-13)

**Note:** Version bump only for package @kongponents/kooltip





## [4.0.2](https://github.com/Kong/kongponents/compare/v4.0.1...v4.0.2) (2021-09-09)

**Note:** Version bump only for package @kongponents/kooltip





## [4.0.1](https://github.com/Kong/kongponents/compare/v4.0.0...v4.0.1) (2021-09-07)

**Note:** Version bump only for package @kongponents/kooltip





# 4.0.0 (2021-09-03)


* feat(kooltip)!: refactor to kpop and restyle (#283) ([ede3f3c](https://github.com/Kong/kongponents/commit/ede3f3c8250559b81def17c24cebb88fd28e9055)), closes [#283](https://github.com/Kong/kongponents/issues/283)


### Bug Fixes

* **dist:** dist to all packages ([#152](https://github.com/Kong/kongponents/issues/152)) ([217c46b](https://github.com/Kong/kongponents/commit/217c46b02afe960f8fa597ba77410dcf034b1716))


### BREAKING CHANGES

* props and styles completely updated





# [1.0.0](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.8...@kongponents/kooltip@1.0.0) (2020-12-18)


* feat(kooltip)!: refactor to kpop and restyle (#283) ([ede3f3c](https://github.com/Kong/kongponents/commit/ede3f3c8250559b81def17c24cebb88fd28e9055)), closes [#283](https://github.com/Kong/kongponents/issues/283)


### BREAKING CHANGES

* props and styles completely updated





## [0.2.8](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.7...@kongponents/kooltip@0.2.8) (2020-03-06)

**Note:** Version bump only for package @kongponents/kooltip





## [0.2.7](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.5...@kongponents/kooltip@0.2.7) (2020-01-16)

**Note:** Version bump only for package @kongponents/kooltip





## [0.2.5](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.4...@kongponents/kooltip@0.2.5) (2019-11-12)

**Note:** Version bump only for package @kongponents/kooltip





## [0.2.4](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.3...@kongponents/kooltip@0.2.4) (2019-11-11)

**Note:** Version bump only for package @kongponents/kooltip





## [0.2.3](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.2...@kongponents/kooltip@0.2.3) (2019-11-08)


### Bug Fixes

* **dist:** dist to all packages ([#152](https://github.com/Kong/kongponents/issues/152)) ([217c46b](https://github.com/Kong/kongponents/commit/217c46b02afe960f8fa597ba77410dcf034b1716))





## [0.2.2](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.1...@kongponents/kooltip@0.2.2) (2019-11-08)

**Note:** Version bump only for package @kongponents/kooltip





## [0.2.1](https://github.com/Kong/kongponents/compare/@kongponents/kooltip@0.2.0...@kongponents/kooltip@0.2.1) (2019-11-07)

**Note:** Version bump only for package @kongponents/kooltip
