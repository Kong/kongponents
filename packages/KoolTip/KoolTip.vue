<template>
  <div
    :aria-hidden="isHovering ? 'true' : 'false'"
    :data-message="message"
    :class="positionClass"
    :style="{'text-align': alignment}"
    class="k-tooltip"
    role="tooltip"
    @mouseout="isHovering(false)"
    @mouseover="isHovering(true)">
    <slot/>
  </div>
</template>

<script>
export default {
  name: 'KoolTip',
  props: {
    /**
     * Message to show in tooltip
     */
    message: {
      type: String,
      required: true
    },
    /**
     * Define which side the tooltip displays<br>
     * 'top' | 'bottom' | 'left' | 'right'
     */
    position: {
      type: String,
      default: 'top',
      validator: function (value) {
        return [
          'top',
          'bottom',
          'left',
          'right'
        ].indexOf(value) !== -1
      }
    },
    /**
     * Sets the text alignment of the tooltip<br>
     * 'left' | 'center' | 'right'
     */
    alignment: {
      type: String,
      default: 'left',
      validator: function (value) {
        return [
          'left',
          'right',
          'center'
        ].indexOf(value) !== -1
      }
    }
  },

  data () {
    return {
      hovering: false
    }
  },

  computed: {
    positionClass () {
      return `k-tooltip-${this.position}`
    }
  },

  methods: {
    isHovering (isHovering) {
      this.hovering = isHovering
    }
  }
}
</script>

<style>
.k-tooltip {
  display: inline-block;
  position: relative;
  cursor: pointer;
}
.k-tooltip:before,
.k-tooltip:after {
  position: absolute;
  visibility: hidden;
  opacity: 0;
  pointer-events: none;
  z-index: 1090;
  transition: all 0.15s cubic-bezier(0.5, 1, 0.25, 1);
}
.k-tooltip:before {
  min-width: 275px;
  padding: .5rem .75rem;
  font-size: .875rem;
  font-weight: normal;
  border-radius: 3px;
  background: #fff;
  box-shadow: 0 0 12px 2px rgba(0, 0, 0, 0.15);
  content: attr(data-message);
}
.k-tooltip:after {
  width: 0;
  border: 8px solid transparent;
  content: "";
}
.k-tooltip:hover:before,
.k-tooltip:hover:after {
  visibility: visible;
  opacity: 1;
}

  /* position styling */
.k-tooltip.k-tooltip-top:before,
.k-tooltip.k-tooltip-top:after {
  bottom: 100%;
  transform: translateX(-50%);
}
.k-tooltip.k-tooltip-top:before {
  left: 100%;
  margin-bottom: 5px;
}
.k-tooltip.k-tooltip-top:after {
  left: 50%;
  border-top: 8px solid #fff;
  border-bottom: none;
  margin-bottom: -3px;
}
.k-tooltip.k-tooltip-top:hover:before,
.k-tooltip.k-tooltip-top:hover:after {
  transform: translateX(-50%) translateY(-5px);
}

.k-tooltip.k-tooltip-right:before,
.k-tooltip.k-tooltip-right:after {
  top: 50%;
  left: 100%;
  transform: translateY(-50%)
}
.k-tooltip.k-tooltip-right:before {
  margin-left: 5px;
}
.k-tooltip.k-tooltip-right:after {
  border-right: 8px solid #fff;
  border-left: none;
  margin-left: -3px;
}
.k-tooltip.k-tooltip-right:hover:before,
.k-tooltip.k-tooltip-right:hover:after {
  transform: translateX(5px) translateY(-50%);
}

.k-tooltip.k-tooltip-bottom:before,
.k-tooltip.k-tooltip-bottom:after {
  top: 100%;
  transform: translateX(-50%);
}
.k-tooltip.k-tooltip-bottom:before {
  left: 100%;
  margin-top: 5px;
}
.k-tooltip.k-tooltip-bottom:after {
  left: 50%;
  border-bottom: 8px solid #fff;
  border-top: none;
  margin-top: -3px;
}
.k-tooltip.k-tooltip-bottom:hover:before,
.k-tooltip.k-tooltip-bottom:hover:after {
  transform: translateX(-50%) translateY(5px);
}

.k-tooltip.k-tooltip-left:before,
.k-tooltip.k-tooltip-left:after {
  top: 50%;
  right: 100%;
  transform: translateY(-50%);
}
.k-tooltip.k-tooltip-left:before {
  margin-right: 5px;
}
.k-tooltip.k-tooltip-left:after {
  border-left: 8px solid #fff;
  border-right: none;
  margin-right: -3px;
}
.k-tooltip.k-tooltip-left:hover:before,
.k-tooltip.k-tooltip-left:hover:after {
  transform: translateX(-5px) translateY(-50%);
}
</style>