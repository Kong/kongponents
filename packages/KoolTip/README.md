```js
const attributes = {
  description: 'KoolTip description text can be as short or as long as you wish.',
}
<KoolTip :message="attributes.description">
  Hover over me to see tooltip
</KoolTip>
```