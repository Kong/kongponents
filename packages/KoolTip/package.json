{
  "name": "@kongponents/kooltip",
  "version": "0.2.0",
  "description": "A tooltip component",
  "main": "KoolTip.vue",
  "scripts": {
    "test": "jest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme"
}
