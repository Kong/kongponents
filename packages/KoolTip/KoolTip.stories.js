import { storiesOf } from '../../storybook/node_modules/@storybook/vue'

import KoolTip from './KoolTip.vue'

storiesOf('KoolTip', module)
  .add('Button with default positioning', () => ({
    components: { KoolTip },
    template: `<KoolTip style="margin-top:50px;margin-left:100px;" message="I am some sample text with default positioning!"><button>button</button></KoolTip>`
  }), {info: {}})
  .add('Button with right positioning', () => ({
    components: { KoolTip },
    template: `<KoolTip position="right" message="I am some sample text with right positioning!"><button>button</button></KoolTip>`
  }), {info: {}})
