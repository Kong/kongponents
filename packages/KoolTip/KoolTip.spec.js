import { mount } from '@vue/test-utils'
import KoolTip from '@/KoolTip/KoolTip'

const positions = ['top', 'right', 'bottom', 'left']
const alignments = ['left', 'center', 'right']

const rendersCorrectPosition = (variant) => {
  it(`renders tooltip to the ${variant} side`, () => {
    const wrapper = mount(KoolTip, {
      propsData: {
        'position': variant,
        'message': `I'm on the ${variant} side!`
      }
    })

    expect(wrapper.find('.k-tooltip').classes()).toContain(`k-tooltip-${variant}`)
  })
}

let rendersCorrectAlignment = (variant) => {
  it(`renders tooltip text aligned to the ${variant}`, () => {
    const wrapper = mount(KoolTip, {
      propsData: {
        'alignment': variant,
        'message': `I'm aligned to the ${variant}!`
      }
    })

    expect(wrapper.element.style['text-align']).toEqual(variant)
  })
}

describe('KoolTip', () => {
  positions.map(p => rendersCorrectPosition(p))
  alignments.map(a => rendersCorrectAlignment(a))

  it('matches snapshot', () => {
    const wrapper = mount(KoolTip, {
      propsData: {
        'message': `I'm inside the tooltip!`
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})