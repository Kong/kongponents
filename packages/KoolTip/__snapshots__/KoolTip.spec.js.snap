// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KoolTip matches snapshot 1`] = `<div aria-hidden="true" data-message="I'm inside the tooltip!" role="tooltip" class="k-tooltip k-tooltip-top" style="text-align: left;"></div>`;
